homeassistant/helpers/template.py

```diff
@@ -653,6 +655,71 @@ def distance(hass, *args):
     )
 
 
+def coordinates(
+    hass: HomeAssistantType, entity_id: str, recursion_history: Optional[list] = None
+) -> Optional[str]:
+    """Get the location from the entity state or attributes."""
+    entity = _resolve_state(hass, entity_id)
+
+    if entity is None:
+        _LOGGER.error("Unable to find entity %s", entity_id)
+        return None
+
+    # Check if the entity has location attributes
+    if loc_helper.has_location(entity):
+        return _get_location_from_attributes(entity)
+
+    # Check if device is in a zone
+    zone_entity = _resolve_state(hass, f"zone.{entity.state}")
+    if loc_helper.has_location(zone_entity):  # type: ignore
+        _LOGGER.debug(
+            "%s is in %s, getting zone location", entity_id, zone_entity.entity_id  # type: ignore
+        )
+        return _get_location_from_attributes(zone_entity)  # type: ignore
+
+    # Resolve nested entity
+    if recursion_history is None:
+        recursion_history = []
+    recursion_history.append(entity_id)
+    if entity.state in recursion_history:
+        _LOGGER.error(
+            "Circular Reference detected. The state of %s has already been checked.",
+            entity.state,
+        )
+        return None
+    _LOGGER.debug("Getting nested entity for state: %s", entity.state)
+    nested_entity = _resolve_state(hass, entity.state)
+    if nested_entity is not None:
+        _LOGGER.debug("Resolving nested entity_id: %s", entity.state)
+        return coordinates(hass, entity.state, recursion_history)
+
+    # Check if state is valid coordinate set
+    if _entity_state_is_valid_coordinate_set(entity.state):
+        return entity.state
+
+    _LOGGER.error(
+        "The state of %s is not a valid set of coordinates: %s", entity_id, entity.state
+    )
+    return None
+
+
+def _entity_state_is_valid_coordinate_set(state: str) -> bool:
```

Comment:
This function is only used once, you can also use:
```
try:
   cv.gps(state.split(","))
except vol.Invalid:
   _LOGGER.error(...)
else:
  return entity.state
```

Reply:
Removed the function